lib/Serialization/SerializeSIL.cpp

```diff
@@ -2276,14 +2284,17 @@ void SILSerializer::writeSILBlock(const SILModule *SILMod) {
   const DeclContext *assocDC = SILMod->getAssociatedContext();
   assert(assocDC && "cannot serialize SIL without an associated DeclContext");
   for (const SILVTable &vt : SILMod->getVTables()) {
-    if ((ShouldSerializeAll || SILMod->getOptions().SILSerializeVTables) &&
+    if ((ShouldSerializeAll ||
```

Comment:
I think you can remove ShouldSerializeAll also

Reply:
Do you mean only in this condition or in the whole file? In spite of its name, this was not set by `-sil-serialize-all`, but by something different.

Reply:
It looks like when emitting SIB, we set it unconditionally, but otherwise it is keyed off -sil-serialize-all. I suggest removing it as well as serializationOpts.SerializeAllSIL.

Reply:
I see some other usages of this:

```
Slavas-MacBook-Pro:swift slava$ git grep SerializeAllSIL
include/swift/Serialization/SerializationOptions.h:    bool SerializeAllSIL = false;
lib/FrontendTool/FrontendTool.cpp:      serializationOpts.SerializeAllSIL = true;
lib/FrontendTool/FrontendTool.cpp:      serializationOpts.SerializeAllSIL = true;
lib/FrontendTool/FrontendTool.cpp:      serializationOpts.SerializeAllSIL =
lib/Serialization/Serialization.cpp:    S.writeSIL(SILMod, options.SerializeAllSIL);
tools/sil-func-extractor/SILFunctionExtractor.cpp:    serializationOpts.SerializeAllSIL = true;
tools/sil-opt/SILOpt.cpp:    serializationOpts.SerializeAllSIL = true;
```

I suspect they're all bogus. Please just rip it out, and since sil-func-extractor and sil-opt are not part of the usual developer flow, any tests that fail as a result are probably bogus.